pkg/providers/vsphere/reconciler/reconciler.go

```diff
@@ -113,181 +96,57 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *an
 		}, nil
 	}
 
-	machineConfigMap := map[string]*anywherev1.VSphereMachineConfig{}
-
-	for _, ref := range cluster.MachineConfigRefs() {
-		machineConfig := &anywherev1.VSphereMachineConfig{}
-		machineConfigName := types.NamespacedName{Namespace: cluster.Namespace, Name: ref.Name}
-		if err := r.client.Get(ctx, machineConfigName, machineConfig); err != nil {
-			return controller.Result{}, err
-		}
-		machineConfigMap[ref.Name] = machineConfig
-	}
-
 	log.V(4).Info("Fetching cluster spec")
 	specWithBundles, err := c.BuildSpec(ctx, clientutil.NewKubeClient(r.client), cluster)
 	if err != nil {
 		return controller.Result{}, err
 	}
 
-	vsphereClusterSpec := vsphere.NewSpec(specWithBundles, machineConfigMap, dataCenterConfig)
-
-	if err := r.validator.ValidateClusterMachineConfigs(ctx, vsphereClusterSpec); err != nil {
+	if err := r.validateMachineConfigs(ctx, cluster, specWithBundles, dataCenterConfig); err != nil {
```

Comment:
Could we make the validation another phase in the runner?

Reply:
Yes, with the other changes that you suggested, we can make this another phase